index.js

```diff
@@ -109,6 +109,12 @@ export function execa(file, args, options) {
 	spawned.kill = spawnedKill.bind(null, spawned.kill.bind(spawned));
 	spawned.cancel = spawnedCancel.bind(null, spawned, context);
 
+	if (parsed.options.signal) {
+		parsed.options.signal.addEventListener('abort', () => {
```

Comment:
When the `signal` option is passed to `child_process.spawn()`, `spawned.kill()` is already automatically called by Node.js when the `abort` event is emitted.

`spawned.cancel()` also sets `error.isCanceled: true`. However, we could set it instead by replacing the following line:

https://github.com/sindresorhus/execa/blob/e4b929547ade335b0dd9d7afe1a35989e19fbc8a/index.js#L130

By:

```js
    isCanceled: parsed.options.signal && parsed.options.signal.aborted,
```